### PR TITLE
Split ReasourceReader and ResourceWriter

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
The reader will be used commonly in very low level scenarios.
For example, we want to move all corefx to use the reader instead of relying on ResourceManager.
On the other hand, the writer is mainly for tooling scenarios.

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
@gkhanna79, @weshaggard, @stephentoub

**Comment by KrzysztofCwalina** [comment]
I just did an update to this PR. I added the merged RW back after talking to @weshaggard. This is to help in transitioning dependencies. After we transition the dependencies, the RW will be removed.

All the code in the new components is identical to the RW code, i.e. I just added new project files.

File: src/System.Resources.Reader/src/System/Resources/ResourceReader.cs [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+{
+    using System;
+    using System.IO;
+    using System.Globalization;
+    using System.Collections;
+    using System.Text;
+    using System.Threading;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    public sealed class ResourceReader : IDisposable
```
Comment (weshaggard) [inline src/System.Resources.Reader/src/System/Resources/ResourceReader.cs]: I assume this copied from the old code but according to our guidelines the usings should be at the top outside the namespace.

File: src/System.Resources.Reader/src/System/Resources/ResourceReader.cs [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+                    _store = null;
+                    if (copyOfStore != null)
+                        copyOfStore.Dispose();
+                }
+                _store = null;
+                _namePositions = null;
+
+            }
+        }
+        private void SkipString()
```
Comment (weshaggard) [inline src/System.Resources.Reader/src/System/Resources/ResourceReader.cs]: I can see some other inconsistent formatting where some methods have 2 blank lines and others have 0. The should all have 1. I'm fine with these being cleaned up separate from your split.

File: src/System.Resources.Reader/src/project.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "System.Diagnostics.Contracts": "4.0.0",
+    "System.Diagnostics.Debug": "4.0.10",
+    "System.Diagnostics.Tools": "4.0.0",
+    "System.IO": "4.0.10",
+    "System.Resources.ResourceManager": "4.0.0",
+    "System.Runtime": "4.0.20",
+    "System.Runtime.Extensions": "4.0.10",
+    "System.Text.Encoding": "4.0.10",
+    "System.Threading": "4.0.10"
+  },
```
Comment (weshaggard) [inline src/System.Resources.Reader/src/project.json]: How many of these dependencies need to be eliminated if you want to add this to the standard platform.

**Comment by weshaggard** [comment]
Beside some comments on the style which are orthogonal to your split this looks good to me.

File: src/System.Resources.Reader/src/System/Resources/ResourceReader.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.Resources.Reader/src/System/Resources/ResourceReader.cs]: Yeah, it's just copied. I will be doing additional cleanup later. I wanted this PR to include just the split, so there is no confusion as to what code changes are actually being made.

File: src/System.Resources.Reader/src/project.json (hunk 3 above)
Comment (KrzysztofCwalina) [inline src/System.Resources.Reader/src/project.json]: That's the work I will be doing going forward.

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test this please

**Pull request merged**